### PR TITLE
ERMS-3205 + ERMS-3204

### PR DESCRIPTION
**Pull request by Moetez16**

### CODE REVIEW

File: server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy [hunk 1]
```diff
@@ -2275,4 +2276,55 @@ class WorkflowController {
     log.debug('triggerSourceUpdate() done - redirecting')
     redirect(url: request.getHeader('referer'));
   }
+
+  @Secured(['ROLE_USER', 'IS_AUTHENTICATED_FULLY'])
+  private def triggerSourceUpdateAllTitles(packages_to_update) {
```
Comment (philboeselager) [inline server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy]: To keep the source maintainable, better write:

    private def triggerSourceUpdateAllTitles(packages_to_update) {
        triggerSourceUpdate(packages_to_update, true)
    }

    private def triggerSourceUpdate(packages_to_update, boolean allTitles=false) {
        ...
        if (allTitles){
            // put your all titles code here
        }
    }

**Review by philboeselager — COMMENTED** [review]
Try to avoid multiple versions of the same code. Everything else looks fine.

File: server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy (hunk 1 above)
Comment (Moetez16) [reply server/gokbg3/grails-app/controllers/org/gokb/WorkflowController.groovy]: Thx.

**Comment by Moetez16** [comment]
PR geupdatet

**Review by philboeselager — APPROVED** [review]

**Pull request merged**